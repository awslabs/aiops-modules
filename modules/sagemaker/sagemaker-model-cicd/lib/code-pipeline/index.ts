export * from './mlops-code-pipeline-stack';
export * from './mlops-code-pipeline-stack-props';
export * from './model-build-support/model-build-support-stack';
export * from './model-build-support/model-build-support-stage';
export * from './model-build/model-build-code-pipeline-stack';
export * from './model-build/model-build-code-pipeline-stage';
export * from './model-deploy/model-deploy-code-pipeline-stack';
export * from './model-deploy/model-deploy-code-pipeline-stage';
export * from './model-endpoint/model-endpoint-stack';
export * from './model-endpoint/model-endpoint-stage';
