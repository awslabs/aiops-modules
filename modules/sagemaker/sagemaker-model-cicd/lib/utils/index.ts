export * from './create-resource-name';
export * from './get-seedfarmer-env-vars';
export * from './pipeline-artifacts-bucket';
