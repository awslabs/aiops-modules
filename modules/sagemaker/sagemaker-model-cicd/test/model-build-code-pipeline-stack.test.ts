import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { ModelBuildCodePipelineStack } from '../lib';

test('ModelBuildCodePipelineStack to match snapshot', () => {
  const app = new cdk.App();
  const stack = new ModelBuildCodePipelineStack(app, 'MyTestStack', {
    projectName: 'test-project',
    sourceBranch: 'main',
    modelBuildEnvironment: {
      name: 'model-build',
      account: '123456789012',
      region: 'us-east-1',
      type: 'dev',
    },
    modelBuildSourceRepoName: 'model-build-repo',
    sagemakerArtifactsBucketName: 'sagemaker-artifacts-bucket',
    sagemakerExecutionRoleName: 'sagemaker-execution-role',
    codeBuildAssumeRoleName: 'codebuild-assume-role',
    modelPackageGroupName: 'model-package-group',
  });
  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
