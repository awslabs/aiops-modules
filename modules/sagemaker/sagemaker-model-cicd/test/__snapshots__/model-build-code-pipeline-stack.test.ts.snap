// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelBuildCodePipelineStack to match snapshot 1`] = `
{
  "Mappings": {
    "LatestNodeRuntimeMap": {
      "af-south-1": {
        "value": "nodejs20.x",
      },
      "ap-east-1": {
        "value": "nodejs20.x",
      },
      "ap-northeast-1": {
        "value": "nodejs20.x",
      },
      "ap-northeast-2": {
        "value": "nodejs20.x",
      },
      "ap-northeast-3": {
        "value": "nodejs20.x",
      },
      "ap-south-1": {
        "value": "nodejs20.x",
      },
      "ap-south-2": {
        "value": "nodejs20.x",
      },
      "ap-southeast-1": {
        "value": "nodejs20.x",
      },
      "ap-southeast-2": {
        "value": "nodejs20.x",
      },
      "ap-southeast-3": {
        "value": "nodejs20.x",
      },
      "ap-southeast-4": {
        "value": "nodejs20.x",
      },
      "ca-central-1": {
        "value": "nodejs20.x",
      },
      "cn-north-1": {
        "value": "nodejs18.x",
      },
      "cn-northwest-1": {
        "value": "nodejs18.x",
      },
      "eu-central-1": {
        "value": "nodejs20.x",
      },
      "eu-central-2": {
        "value": "nodejs20.x",
      },
      "eu-north-1": {
        "value": "nodejs20.x",
      },
      "eu-south-1": {
        "value": "nodejs20.x",
      },
      "eu-south-2": {
        "value": "nodejs20.x",
      },
      "eu-west-1": {
        "value": "nodejs20.x",
      },
      "eu-west-2": {
        "value": "nodejs20.x",
      },
      "eu-west-3": {
        "value": "nodejs20.x",
      },
      "il-central-1": {
        "value": "nodejs20.x",
      },
      "me-central-1": {
        "value": "nodejs20.x",
      },
      "me-south-1": {
        "value": "nodejs20.x",
      },
      "sa-east-1": {
        "value": "nodejs20.x",
      },
      "us-east-1": {
        "value": "nodejs20.x",
      },
      "us-east-2": {
        "value": "nodejs20.x",
      },
      "us-gov-east-1": {
        "value": "nodejs18.x",
      },
      "us-gov-west-1": {
        "value": "nodejs18.x",
      },
      "us-iso-east-1": {
        "value": "nodejs18.x",
      },
      "us-iso-west-1": {
        "value": "nodejs18.x",
      },
      "us-isob-east-1": {
        "value": "nodejs18.x",
      },
      "us-west-1": {
        "value": "nodejs20.x",
      },
      "us-west-2": {
        "value": "nodejs20.x",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "BuildPipelineB0A659C0": {
      "DependsOn": [
        "BuildPipelineRoleDefaultPolicy47F6D8A3",
        "BuildPipelineRole7BF83B9B",
      ],
      "Properties": {
        "ArtifactStore": {
          "EncryptionKey": {
            "Id": {
              "Fn::GetAtt": [
                "PipelineArtifactsBucketKms876A3C5B",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": {
            "Ref": "PipelineArtifactBucketD127CCF6",
          },
          "Type": "S3",
        },
        "Name": "test-project-main-build",
        "RoleArn": {
          "Fn::GetAtt": [
            "BuildPipelineRole7BF83B9B",
            "Arn",
          ],
        },
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeCommit",
                  "Version": "1",
                },
                "Configuration": {
                  "BranchName": "main",
                  "PollForSourceChanges": false,
                  "RepositoryName": "model-build-repo",
                },
                "Name": "Source",
                "OutputArtifacts": [
                  {
                    "Name": "GitSource",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "BuildPipelineSourceCodePipelineActionRole2D2EC58B",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "ProjectName": {
                    "Ref": "SMPipelineBuild0995B116",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "GitSource",
                  },
                ],
                "Name": "SMPipeline",
                "RoleArn": {
                  "Fn::GetAtt": [
                    "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRole7AC29603",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "StartModelBuild",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "BuildPipelineEventsRoleA981CBE7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "events.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineEventsRoleDefaultPolicyEEDBB7FE": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "codepipeline:StartPipelineExecution",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codepipeline:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":",
                    {
                      "Ref": "BuildPipelineB0A659C0",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineEventsRoleDefaultPolicyEEDBB7FE",
        "Roles": [
          {
            "Ref": "BuildPipelineEventsRoleA981CBE7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineRole7BF83B9B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineRoleDefaultPolicy47F6D8A3": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactBucketD127CCF6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactBucketD127CCF6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineArtifactsBucketKms876A3C5B",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "BuildPipelineSourceCodePipelineActionRole2D2EC58B",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRole7AC29603",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineRoleDefaultPolicy47F6D8A3",
        "Roles": [
          {
            "Ref": "BuildPipelineRole7BF83B9B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineSourceCodePipelineActionRole2D2EC58B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineSourceCodePipelineActionRoleDefaultPolicy853F1EB8": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactBucketD127CCF6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactBucketD127CCF6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineArtifactsBucketKms876A3C5B",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "codecommit:GetBranch",
                "codecommit:GetCommit",
                "codecommit:UploadArchive",
                "codecommit:GetUploadArchiveStatus",
                "codecommit:CancelUploadArchive",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codecommit:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":model-build-repo",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineSourceCodePipelineActionRoleDefaultPolicy853F1EB8",
        "Roles": [
          {
            "Ref": "BuildPipelineSourceCodePipelineActionRole2D2EC58B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRole7AC29603": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRoleDefaultPolicy875CDFCB": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "SMPipelineBuild0995B116",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRoleDefaultPolicy875CDFCB",
        "Roles": [
          {
            "Ref": "BuildPipelineStartModelBuildSMPipelineCodePipelineActionRole7AC29603",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "BuildRepoMyTestStackBuildPipelineE3FA9E86mainEventRule669AD1C5": {
      "Properties": {
        "EventPattern": {
          "detail": {
            "event": [
              "referenceCreated",
              "referenceUpdated",
            ],
            "referenceName": [
              "main",
            ],
          },
          "detail-type": [
            "CodeCommit Repository State Change",
          ],
          "resources": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":codecommit:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":model-build-repo",
                ],
              ],
            },
          ],
          "source": [
            "aws.codecommit",
          ],
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":codepipeline:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":",
                  {
                    "Ref": "BuildPipelineB0A659C0",
                  },
                ],
              ],
            },
            "Id": "Target0",
            "RoleArn": {
              "Fn::GetAtt": [
                "BuildPipelineEventsRoleA981CBE7",
                "Arn",
              ],
            },
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "CodeBuildPolicy9FEF6D56": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": "arn:aws:iam::123456789012:role/codebuild-assume-role",
              "Sid": "AssumeSageMakerAccessRole",
            },
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "LogsAccess",
            },
            {
              "Action": "cloudwatch:PutMetricData",
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "MetricsAccess",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CodeBuildPolicy9FEF6D56",
        "Roles": [
          {
            "Ref": "CodeBuildRole728CBADE",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CodeBuildRole728CBADE": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/service-role/",
      },
      "Type": "AWS::IAM::Role",
    },
    "CodeBuildRoleDefaultPolicy829527DE": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "SMPipelineBuild0995B116",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      {
                        "Ref": "SMPipelineBuild0995B116",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    {
                      "Ref": "SMPipelineBuild0995B116",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactBucketD127CCF6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactBucketD127CCF6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineArtifactsBucketKms876A3C5B",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "PipelineArtifactsBucketKms876A3C5B",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CodeBuildRoleDefaultPolicy829527DE",
        "Roles": [
          {
            "Ref": "CodeBuildRole728CBADE",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "faa95a81ae7d7373f3e1f242268f904eb748d8d0fdd306e8a6fe515a1905a7d6.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "PipelineArtifactBucketD127CCF6",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": {
          "Fn::FindInMap": [
            "LatestNodeRuntimeMap",
            {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "PipelineArtifactBucketAutoDeleteObjectsCustomResource00FE3AE7": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "PipelineArtifactBucketPolicyC0FB57EC",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "PipelineArtifactBucketD127CCF6",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "PipelineArtifactBucketD127CCF6": {
      "DeletionPolicy": "Delete",
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-S1",
              "reason": "The bucket stores pipeline artifacts, no logging required.",
            },
          ],
        },
      },
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "BucketKeyEnabled": true,
              "ServerSideEncryptionByDefault": {
                "KMSMasterKeyID": {
                  "Fn::GetAtt": [
                    "PipelineArtifactsBucketKms876A3C5B",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "PipelineArtifactBucketPolicyC0FB57EC": {
      "Properties": {
        "Bucket": {
          "Ref": "PipelineArtifactBucketD127CCF6",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactBucketD127CCF6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactBucketD127CCF6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "PipelineArtifactBucketD127CCF6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "PipelineArtifactBucketD127CCF6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "PipelineArtifactsBucketKms876A3C5B": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "EnableKeyRotation": true,
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "SMPipelineBuild0995B116": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "EncryptionKey": {
          "Fn::GetAtt": [
            "PipelineArtifactsBucketKms876A3C5B",
            "Arn",
          ],
        },
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "EnvironmentVariables": [
            {
              "Name": "AWS_REGION",
              "Type": "PLAINTEXT",
              "Value": "us-east-1",
            },
            {
              "Name": "ASSUME_ROLE_ARN",
              "Type": "PLAINTEXT",
              "Value": "arn:aws:iam::123456789012:role/codebuild-assume-role",
            },
            {
              "Name": "ENV_TYPE",
              "Type": "PLAINTEXT",
              "Value": "dev",
            },
            {
              "Name": "PROJECT_NAME",
              "Type": "PLAINTEXT",
              "Value": "test-project",
            },
            {
              "Name": "MODEL_PACKAGE_GROUP_NAME",
              "Type": "PLAINTEXT",
              "Value": "model-package-group",
            },
            {
              "Name": "SAGEMAKER_PIPELINE_NAME",
              "Type": "PLAINTEXT",
              "Value": "test-project-main-build",
            },
            {
              "Name": "SAGEMAKER_PIPELINE_DESCRIPTION",
              "Type": "PLAINTEXT",
              "Value": "test-project model build pipeline for main branch",
            },
            {
              "Name": "SAGEMAKER_PIPELINE_ROLE_ARN",
              "Type": "PLAINTEXT",
              "Value": "arn:aws:iam::123456789012:role/service-role/sagemaker-execution-role",
            },
            {
              "Name": "ARTIFACT_BUCKET",
              "Type": "PLAINTEXT",
              "Value": "sagemaker-artifacts-bucket",
            },
            {
              "Name": "ARTIFACT_BUCKET_KMS_KEY",
              "Type": "PLAINTEXT",
              "Value": "alias/sagemaker-artifacts-bucket",
            },
          ],
          "Image": "aws/codebuild/standard:7.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": "test-project-main-build",
        "ServiceRole": {
          "Fn::GetAtt": [
            "CodeBuildRole728CBADE",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "buildspec.yml",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
